domains/client/domain-operator/src/fraud_proof.rs

```diff
@@ -191,11 +209,105 @@ where
         )
         .ok_or(FraudProofError::FailToGenerateProofOfInclusion)?;
 
+        let invalid_bundles_fraud_proof_type = match invalid_type {
+            InvalidBundleType::UndecodableTx(extrinsic_index) => {
+                InvalidBundlesFraudProofType::UndecodableTx(extrinsic_index)
+            }
+            InvalidBundleType::OutOfRangeTx(extrinsic_index) => {
+                InvalidBundlesFraudProofType::OutOfRangeTx(extrinsic_index)
+            }
+            InvalidBundleType::IllegalTx(expected_extrinsic_index) => {
+                if expected_extrinsic_index as usize >= bundle.extrinsics.len() {
+                    return Err(FraudProofError::OutOfBoundsExtrinsicIndex {
+                        index: expected_extrinsic_index as usize,
+                        max: bundle.extrinsics.len(),
+                    });
+                }
+
+                let domain_block_parent_hash = *self
+                    .client
+                    .header(local_receipt.domain_block_hash)?
+                    .ok_or_else(|| {
+                        FraudProofError::Blockchain(sp_blockchain::Error::MissingHeader(format!(
+                            "{:?}",
+                            local_receipt.domain_block_hash
+                        )))
+                    })?
+                    .parent_hash();
+
+                let domain_block_parent_number = self
+                    .client
+                    .block_number_from_id(&BlockId::Hash(domain_block_parent_hash))?
+                    .ok_or_else(|| {
+                        FraudProofError::Blockchain(sp_blockchain::Error::Backend(format!(
+                            "unable to get block number for domain block:{:?}",
+                            domain_block_parent_hash
+                        )))
+                    })?;
+
+                let mut runtime_api_instance = self.client.runtime_api();
+                runtime_api_instance.record_proof();
+                let proof_recorder = runtime_api_instance
+                    .proof_recorder()
+                    .expect("we enabled proof recording just above; qed");
+
+                let mut block_extrinsics = vec![];
+                for (extrinsic_index, extrinsic) in bundle.extrinsics.iter().enumerate() {
```

Comment:
Same here, we do not need to verify all the extrinsic should stop at `expected_extrinsic_index`